spec/support/features/session_helper.rb

```diff
@@ -570,6 +570,21 @@ def stub_piv_cac_service
           body: CGI.unescape(request.body.sub(/^token=/, '')),
         }
       end
+
+      stub_request(:post, 'piv.example.com').
+        with(query: hash_including('nonce', 'redirect_uri')).
+        to_return do |request|
+          query = CGI.parse(URI.parse(request.uri).query).transform_values(&:first)
+          {
+            status: 302,
+            headers: {
+              location: UriService.add_params(
+                query['redirect_uri'],
```

Comment:
```suggestion
          query = UriService.params(request.uri)
          {
            status: 302,
            headers: {
              location: UriService.add_params(
                query['redirect_uri'],
```

Reply:
this uses `Rack::Util.parse_nested_query` which returns params as single items instead of arrays by default

Reply:
Ah dang, I was already using `UriService` and totally missed that it had a helper for the parsing aspect! 🤦